compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -726,21 +726,24 @@ fn token_kind(token_kind: TokenKind) -> impl NoirParser<Token> {
 
 fn path() -> impl NoirParser<Path> {
     let idents = || ident().separated_by(just(Token::DoubleColon)).at_least(1);
-    let make_path = |kind| move |segments| Path { segments, kind };
+    let make_path = |kind| move |segments, span| Path { segments, kind, span };
 
     let prefix = |key| keyword(key).ignore_then(just(Token::DoubleColon));
-    let path_kind = |key, kind| prefix(key).ignore_then(idents()).map(make_path(kind));
+    let path_kind = |key, kind| {
+        spanned(prefix(key).ignore_then(idents()))
+            .map(move |(segments, span)| make_path(kind)(segments, span))
```

Comment:
Should be able to use `map_with_span` instead of `spanned` here
```suggestion
       prefix(key).ignore_then(idents())
            .map_with_span(move |(segments, span)| make_path(kind)(segments, span))
```